exporter/elasticsearchexporter/util.go

```diff
@@ -6,11 +6,49 @@ package elasticsearchexporter // import "github.com/open-telemetry/opentelemetry
 import (
 	"bytes"
 	"fmt"
+	"strings"
 	"time"
+	"unicode"
 
 	"github.com/lestrrat-go/strftime"
 )
 
+const (
+	maxDataStreamBytes       = 100
+	disallowedNamespaceRunes = "\\/*?\"<>| ,#:"
+	disallowedDatasetRunes   = "-\\/*?\"<>| ,#:"
+)
+
+// Sanitize the datastream fields (dataset, namespace) to apply restrictions
+// as outlined in https://www.elastic.co/guide/en/ecs/current/ecs-data_stream.html
+func sanitizeDataStreamDataset(field string) string {
+	field = strings.Map(replaceReservedRune(disallowedDatasetRunes), field)
+	if len(field) > maxDataStreamBytes {
+		return field[:maxDataStreamBytes]
+	}
+
+	return field
+}
+
+// Sanitize the datastream fields (dataset, namespace) to apply restrictions
+// as outlined in https://www.elastic.co/guide/en/ecs/current/ecs-data_stream.html
+func sanitizeDataStreamNamespace(field string) string {
+	field = strings.Map(replaceReservedRune(disallowedNamespaceRunes), field)
+	if len(field) > maxDataStreamBytes {
+		return field[:maxDataStreamBytes]
+	}
+	return field
+}
```

Comment:
I missed this in the apm-data PR. It'll be clearer if you can refactor this into 3 funcs, `sanitizeDataStreamDataset` and `sanitizeDataStreamNamespace` with the original signature, which calls a new `sanitizeDataStreamField(field, disallowed)` func that does all the work. Like currying in javascript.

This also enables `replaceReservedRune` func to be inlined